### PR TITLE
fix: Add name to source request for ADE [DHIS2-13105]

### PR DESCRIPTION
**Pull request by larshelge**
Adds a mandatory `name` property and an optional `visualization` property to the source request of aggregate data exchange. Adds Javadoc to data model and service methods.

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java [hunk 1]
```diff
@@ -172,6 +173,19 @@ public static List<PeriodType> getAvailablePeriodTypes()
...
 
+    /**
+     * Returns an immutable list of the names of all period types in their
+     * natural order.
+     *
+     * @return an immutable list of the names of all period types.
+     */
+    public static List<String> getAvailablePeriodTypeNames()
+    {
+        return List.copyOf( PERIOD_TYPES.stream()
```
Comment (jbee) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java:184]: unnecessary `copyOf`. `collect` already create a new list.

**Review by jbee — APPROVED** [review]
Looks good overall. Just the unnecessary copying.

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java (hunk 1 above)
Comment (jbee) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java:184]: For that there is `toUnmodifyableList()` collector.

Comment (larshelge) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/PeriodType.java:184]: Right, I will update it.

**Pull request merged**